File: algorithms/TwoSum.java
package algorithms;

import java.util.*;

public class TwoSum {
    public static int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> map = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            int complement = target - nums[i];
            if (map.containsKey(complement)) {
                return new int[] { map.get(complement), i };
            }
            map.put(nums[i], i);
        }
        throw new IllegalArgumentException("No two sum solution");
    }

    public static void main(String[] args) {
        int[] nums = { 2, 7, 11, 15 };
        int target = 9;
        int[] result = twoSum(nums, target);
        int[] result1 = twoSum1(nums, target);
        System.out.println("Indices of the two numbers that add up to " + target + ": " + Arrays.toString(result));
        System.out.println("Indices of the two numbers that add up to " + target + ": " + Arrays.toString(result1));
        // output: Indices of the two numbers that add up to 9: [0, 1]
    }

    public static int[] twoSum1(int[] nums, int target) {
        throw new IllegalArgumentException("target is not matched");
    }
}

File: algorithmsJS/TwoSum.js
class TwoSum {
  static twoSum(nums, target) {
    const map = new Map();
    for (let i = 0; i < nums.length; i++) {
      const complement = target - nums[i];
      if (map.has(complement)) {
        return [map.get(complement), i];
      }
      map.set(nums[i], i);
    }
    throw new Error("No two sum solution");
  }

  static twoSum1(nums, target) {
    throw new Error("target is not matched");
  }
}

// Example usage:
const nums = [2, 7, 11, 15];
const target = 9;
try {
  const result = TwoSum.twoSum(nums, target);
  console.log(
    `Indices of the two numbers that add up to ${target}: [${result}]`
  );
} catch (e) {
  console.log(e.message);
}
try {
  const result1 = TwoSum.twoSum1(nums, target);
  console.log(
    `Indices of the two numbers that add up to ${target}: [${result1}]`
  );
} catch (e) {
  console.log(e.message);
}
